lib/string_reverse.rb

```diff
@@ -1,6 +1,23 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: linear, O(n)
+#                  where n is the length of my_string
+# Space complexity: O(1)
+#                   where n is the length of my_string * the number of bytes a
+#                   character requires for space
 def string_reverse(my_string)
-  raise NotImplementedError
+  return nil if my_string == nil
+
+  i = 0

```

Comment:
One more _super-minor_ nit: these variables (`i` and `j`) could have more descriptive names.

(It's mostly a matter of personal preference when you're writing code at a computer. When doing whiteboard interviews though, you'll want to be explicit to reduce the chance you and/or the interviewer mix variables up.)